src/Sulu/Bundle/MediaBundle/Tests/Functional/Controller/CollectionControllerTest.php

```diff
@@ -1210,6 +1228,155 @@ public function testDeleteById()
         $this->assertTrue(isset($response->message));
     }
 
+    public function testDeleteByIdWithChildren(): void
+    {
+        $collection = $this->createCollection($this->collectionType1);
+        $this->addMedia($collection, 1);
+        $collectionId = $collection->getId();
+
+        $child1 = $this->createCollection($this->collectionType1, [], $collection);
+        $this->addMedia($child1, 1);
+
+        $child11 = $this->createCollection($this->collectionType1, [], $child1);
+        $this->addMedia($child11, 1);
+
+        $child2 = $this->createCollection($this->collectionType1, [], $collection);
+        $this->addMedia($child2, 1);
+
+        $this->em->clear();
+
+        $this->client->jsonRequest('DELETE', '/api/collections/' . $collectionId);
+        $this->assertHttpStatusCode(409, $this->client->getResponse());
+
+        $response = \json_decode((string) $this->client->getResponse()->getContent());
+        $this->assertEquals(DeletionImpossibleChildrenException::EXCEPTION_CODE, $response->code);
```

Comment:
Would `assertSame` the whole json without the provided stacktrace e.g.::

```
$jsonData = \json_decode((string) $this->client->getResponse()->getContent());
unset($jsonData['errors']);
$expectedJson = <<<EOT
{
    "..": ".."
}
EOT;

$this->assertSame($expectedJson, \json_encode($jsonData, JSON_PRETTY_PRINT));
```

Reply:
@alexander-schranz What about comparing two arrays instead of two json strings? Comparing two json strings is harder, because variables like `$child1->getMedia()->first()->getId()` can't be used that easily

Reply:
is okay for me at current state